### PR TITLE
[Asset] Add type-hints to public interfaces and classes

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #32179  
| License       | MIT
| Doc PR        | N/A 

This PR adds type hints to all public interfaces of the Asset component.

### CODE REVIEW

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/Asset/PackageInterface.php [hunk 1]
```diff
@@ -21,18 +21,14 @@ interface PackageInterface
...
 
     /**
      * Returns an absolute or root-relative public path.
      *
-     * @param string $path A path
-     *
      * @return string The public path
      */
-    public function getUrl($path);
+    public function getUrl(string $path);
```
Comment (plozmun) [inline src/Symfony/Component/Asset/PackageInterface.php]: same here

File: src/Symfony/Component/Asset/Packages.php [hunk 2]
```diff
@@ -89,7 +86,7 @@ public function getPackage($name = null)
      *
      * @return string The current version
      */
-    public function getVersion($path, $packageName = null)
+    public function getVersion(string $path, string $packageName = null)
```
Comment (plozmun) [inline src/Symfony/Component/Asset/Packages.php]: same here

File: src/Symfony/Component/Asset/PackageInterface.php [hunk 3]
```diff
@@ -21,18 +21,14 @@ interface PackageInterface
     /**
      * Returns the asset version for an asset.
      *
-     * @param string $path A path
-     *
      * @return string The version string
      */
-    public function getVersion($path);
+    public function getVersion(string $path);
```
[the remark this answers is not in the document]
Comment (xabbuh) [reply src/Symfony/Component/Asset/PackageInterface.php]: We cannot add return type hints without breaking things while adding argument type hints is possible (compare https://3v4l.org/Bnhvo with https://3v4l.org/RLid3).

**Review by derrabus — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jschaedl.

**Pull request merged**